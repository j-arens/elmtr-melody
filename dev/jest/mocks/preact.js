module.exports = {
    unmountComponent: jest.fn(),
};
