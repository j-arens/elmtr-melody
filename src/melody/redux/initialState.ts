import { MachineStates } from '@state-machine/type';
import { View } from './type';

export default {
    machine: {
        currentState: 'stopped' as MachineStates,
        lastState: 'stopped' as MachineStates,
    },
    audio: {
        tracks: [],
        currentTrack: 0,
        lastTrack: 0,
        shuffle: false,
        repeat: false,
        volume: 1,
        currentTime: 0,
        timeSync: 0,
        playbackRate: 1,
        filelength: 0,
    },
    ui: {
        view: 'slider' as View,
        gliderIsDragging: false,
        volIsDragging: false,
        showDock: false,
    },
};
