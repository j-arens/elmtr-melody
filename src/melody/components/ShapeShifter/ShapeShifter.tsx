import { View } from '@redux/type';
import SimpleToolbar from '@views/SimpleToolbar/SimpleToolbar';
import Slider from '@views/Slider/Slider';
import Tracklist from '@views/Tracklist/Tracklist';
import { h } from 'preact';

interface Props {
    view: View;
}

export default ({ view }: Props) => {
    switch (view) {
        case 'simple-toolbar': {
            return (
                <SimpleToolbar />
            );
        }
        case 'slider': {
            return <Slider />;
        }
        case 'tracklist': {
            return <Tracklist />;
        }
        default: {
            return <Slider />;
        }
    }
};
