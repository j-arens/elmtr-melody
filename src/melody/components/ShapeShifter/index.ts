import { State } from '@redux/type';
import { connect } from 'preact-redux';
import ShapeShifter, { StateProps } from './ShapeShifter';

const mapStateToProps = (state: State) => ({
    view: state.ui.view,
});

export default connect<StateProps, {}, {}>(mapStateToProps)(ShapeShifter);
