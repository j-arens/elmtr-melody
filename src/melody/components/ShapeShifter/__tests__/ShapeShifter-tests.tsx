import configureStore from '@redux/index';
import initialState from '@redux/initialState';
import { h } from 'preact';
import { Provider } from 'preact-redux';
import { render } from 'preact-render-to-string';
const track = require('@fixtures/track');
import ShapeShifter from '../index';

describe('Melody', () => {
    let state;

    beforeEach(() => {
        state = {
            ...initialState,
            audio: {
                ...initialState.audio,
                tracks: [track],
            },
        };
    });

    it('renders the slider view', () => {
        const testState = {
            ...state,
            ui: {
                ...state.ui,
                view: 'slider',
            },
        };
        const component = render(
            <Provider store={configureStore(testState)}>
                <ShapeShifter />
            </Provider>,
        );
        expect(component).toMatchSnapshot();
    });

    it('renders the simple toolbar view', () => {
        const testState = {
            ...state,
            ui: {
                ...state.ui,
                view: 'simple-toolbar',
            },
        };
        const component = render(
            <Provider store={configureStore(testState)}>
                <ShapeShifter />
            </Provider>,
        );
        expect(component).toMatchSnapshot();
    });
});
