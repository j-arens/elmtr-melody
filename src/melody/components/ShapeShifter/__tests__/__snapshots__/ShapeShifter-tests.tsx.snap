// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Melody renders the simple toolbar view 1`] = `
<div class="melody__viewContainer toolbar melody-c-toolbar" data-cy="toolbar">
  <div class="controls controls--left melody-controls-primary">
    <button data-cy="ctrl prev" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-width melody-c-btn-space playbackCtrl melody-playbackCtrl"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M11 18V6l-8.5 6 8.5 6zm.5-6l8.5 6V6l-8.5 6z"></path>
      </svg>
    </button>
    <button data-playbackRate data-cy="ctrl catalyst" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-width melody-c-btn-space playbackCtrl melody-playbackCtrl melody-controlBar__catalyst"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <g>
          <circle class="melody-playbackCtrl__ring" cx="12" cy="12" r="12"></circle>
          <path d="M8 5v14l11-7z"></path>
        </g>
      </svg>
    </button>
    <button data-cy="ctrl next" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-width melody-c-btn-space playbackCtrl melody-playbackCtrl playbackCtrl--disabled"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M4 18l8.5-6L4 6v12zm9-12v12l8.5-6L13 6z"></path>
      </svg>
    </button>
  </div>
  <div class="preview melody-c-preview-padding melody-c-trackinfo-width melody-c-trackinfo-order"
  data-cy="track-info">
    <div class="preview__wrap melody-c-trackinfo-margin" data-cy="inner-track-info">
      <span class="preview__time preview__time--left melody-c-time-font melody-c-time-shadow melody-c-time-color"
      data-cy="time-elapsed">0:00</span>
      <div class="preview__trackInfo">
        <p class="preview__title melody-c-title-font melody-c-title-shadow melody-c-title-color"
        data-cy="track-title">lol-title</p>
        <span class="melody-c-separator" data-cy="seperator"></span>
        <p class="preview__artist melody-c-artist-font melody-c-artist-shadow melody-c-artist-color"
        data-cy="track-artist">lol-artist</p>
      </div>
      <span class="preview__time preview__time--right melody-c-time-font melody-c-time-shadow melody-c-time-color"
      data-cy="time-left">0:00</span>
    </div>
    <div class="slider elementor-tab-title " style="padding: 8px 0;">
      <div class="slider__body melody-c-scrubber-body-bg-color" style="height: 5px;"
      data-cy="slider-body">
        <div class="slider__backfill melody-c-scrubber-backfill-bg-color" style="transform: translate3d(0%, 0, 0);"
        data-cy="slider-backfill"></div>
      </div>
    </div>
  </div>
  <div class="controls controls--right melody-controls-secondary">
    <div class="melody-c-btn-primary-color melody-c-btn-width melody-c-btn-space volumeCtrl"
    data-cy="vol-ctrl">
      <svg class="volumeCtrl__icon elementor-tab-title" xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24">
        <circle class="volumeCtrl__levelOutline" cx="12" cy="12" r="12"></circle>
        <circle class="volumeCtrl__level" style="stroke-dashoffset: 0;"
        cx="12" cy="12" r="12"></circle>
        <path d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"></path>
      </svg>
    </div>
    <button data-cy="ctrl shuffle" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-on-color melody-c-btn-width melody-c-btn-space playbackCtrl melody-playbackCtrl melody-shuffle"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"></path>
      </svg>
    </button>
    <button data-cy="ctrl repeat" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-on-color melody-c-btn-width melody-c-btn-space playbackCtrl melody-playbackCtrl melody-repeat"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"></path>
      </svg>
    </button>
    <button data-cy="dock-toggle" class="btn dock__toggle melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-width melody-c-btn-space"
    role="button">
      <svg class="dock__toggleIcon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"></path>
      </svg>
    </button>
  </div>
  <div class="dock elementor-element--dock-controls-primary-color" style="pointer-events: none; visibility: hidden; top: 12px; left: 0px;"
  data-cy="dock-controls">
    <span class="dock__arrow elementor-element--dock-controls-arrow-color elementor-element--dock-controls-border-color"
    style="left: -6px;"></span>
    <div class="dock__controlsGroup elementor-element--dock-controls-border-color"
    data-cy="dock-controls-wrap">
      <button data-cy="download" class="btn dock__control elementor-element--dock-controls-secondary-color elementor-element--dock-controls-text-color elementor-element--dock-controls-font"
      role="button">
        <svg class="dock__controlIcon elementor-element--dock-controls-icon-color"
        xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"></path>
        </svg>
        <span class="dock__controlName">Download</span>
      </button>
      <button data-cy="speed-up" class="btn dock__control elementor-element--dock-controls-secondary-color elementor-element--dock-controls-text-color elementor-element--dock-controls-font"
      role="button">
        <svg class="dock__controlIcon elementor-element--dock-controls-icon-color"
        xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"></path>
        </svg>
        <span class="dock__controlName">Speed up</span>
      </button>
      <button data-cy="slow-down" class="btn dock__control elementor-element--dock-controls-secondary-color elementor-element--dock-controls-text-color elementor-element--dock-controls-font"
      role="button">
        <svg class="dock__controlIcon elementor-element--dock-controls-icon-color"
        xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M19 13H5v-2h14v2z"></path>
        </svg>
        <span class="dock__controlName">Slow down</span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Melody renders the slider view 1`] = `
<div class="melody__viewContainer">
  <div class="preview melody-c-preview-bg melody-c-preview-min-height melody-c-preview-padding"
  data-cy="preview">
    <ul class="Slideshow preview__slideshow melody-c-slider-transition melody-c-slider-timing"
    style="transform: translate3d(0%, 0px, 0px);" data-cy="artwork-slider">
      <li class="Slideshow__slide">
        <figure class="Slideshow__image melody-c-artwork-size melody-c-artwork-repeat melody-c-artwork-position melody-c-artwork-attachment melody-c-image-filters"
        style="background-image: url();" data-cy="track-artwork"></figure>
      </li>
    </ul>
    <div class="preview__header">
      <div class="melody-c-vol-primary-color melody-c-vol-hover-color melody-c-vol-width volumeCtrl"
      data-cy="vol-ctrl">
        <svg class="volumeCtrl__icon elementor-tab-title" xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 24 24">
          <circle class="volumeCtrl__levelOutline" cx="12" cy="12" r="12"></circle>
          <circle class="volumeCtrl__level" style="stroke-dashoffset: 0;"
          cx="12" cy="12" r="12"></circle>
          <path d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"></path>
        </svg>
      </div>
      <button data-cy="dock-toggle" class="btn dock__toggle melody-c-dock-toggle-color melody-c-dock-toggle-size"
      role="button">
        <svg class="dock__toggleIcon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"></path>
        </svg>
      </button>
    </div>
    <div class="preview__marquee">
      <p class="melody-c-title-font melody-c-title-shadow melody-c-title-color"
      data-cy="track-title">lol-title</p>
      <p class="melody-c-artist-font melody-c-artist-shadow melody-c-artist-color"
      data-cy="track-artist">lol-artist</p>
    </div>
    <div class="preview__glider">
      <div class="slider elementor-tab-title " style="padding: 8px 0;">
        <div class="slider__body melody-c-scrubber-body-bg-color" style="height: 5px;"
        data-cy="slider-body">
          <div class="slider__backfill melody-c-scrubber-backfill-bg-color" style="transform: translate3d(0%, 0, 0);"
          data-cy="slider-backfill"></div>
        </div>
      </div>
    </div>
    <div class="preview__footer">
      <span class="melody-c-time-font melody-c-time-shadow melody-c-time-color"
      data-cy="time-elapsed">0:00</span>
      <span class="melody-c-time-font melody-c-time-shadow melody-c-time-color"
      data-cy="time-left">0:00</span>
    </div>
  </div>
  <div class="controlbar melody-c-controlbar melody-control-bar" data-cy="controlbar">
    <button data-cy="ctrl shuffle" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-space playbackCtrl melody-playbackCtrl melody-shuffle"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"></path>
      </svg>
    </button>
    <button data-cy="ctrl prev" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-space playbackCtrl melody-playbackCtrl"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M11 18V6l-8.5 6 8.5 6zm.5-6l8.5 6V6l-8.5 6z"></path>
      </svg>
    </button>
    <button data-playbackRate data-cy="ctrl catalyst" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-space playbackCtrl melody-playbackCtrl melody-controlBar__catalyst"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <g>
          <circle class="melody-playbackCtrl__ring" cx="12" cy="12" r="12"></circle>
          <path d="M8 5v14l11-7z"></path>
        </g>
      </svg>
    </button>
    <button data-cy="ctrl next" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-space playbackCtrl melody-playbackCtrl playbackCtrl--disabled"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M4 18l8.5-6L4 6v12zm9-12v12l8.5-6L13 6z"></path>
      </svg>
    </button>
    <button data-cy="ctrl repeat" class="btn melody-c-btn-primary-color melody-c-btn-hover-color melody-c-btn-space playbackCtrl melody-playbackCtrl melody-repeat"
    role="button">
      <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"></path>
      </svg>
    </button>
  </div>
  <div class="dock elementor-element--dock-controls-primary-color" style="pointer-events: none; visibility: hidden; top: 12px; left: 0px;"
  data-cy="dock-controls">
    <span class="dock__arrow elementor-element--dock-controls-arrow-color elementor-element--dock-controls-border-color"
    style="left: -6px;"></span>
    <div class="dock__controlsGroup elementor-element--dock-controls-border-color"
    data-cy="dock-controls-wrap">
      <button data-cy="download" class="btn dock__control elementor-element--dock-controls-secondary-color elementor-element--dock-controls-text-color elementor-element--dock-controls-font"
      role="button">
        <svg class="dock__controlIcon elementor-element--dock-controls-icon-color"
        xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"></path>
        </svg>
        <span class="dock__controlName">Download</span>
      </button>
      <button data-cy="speed-up" class="btn dock__control elementor-element--dock-controls-secondary-color elementor-element--dock-controls-text-color elementor-element--dock-controls-font"
      role="button">
        <svg class="dock__controlIcon elementor-element--dock-controls-icon-color"
        xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"></path>
        </svg>
        <span class="dock__controlName">Speed up</span>
      </button>
      <button data-cy="slow-down" class="btn dock__control elementor-element--dock-controls-secondary-color elementor-element--dock-controls-text-color elementor-element--dock-controls-font"
      role="button">
        <svg class="dock__controlIcon elementor-element--dock-controls-icon-color"
        xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
          <path d="M19 13H5v-2h14v2z"></path>
        </svg>
        <span class="dock__controlName">Slow down</span>
      </button>
    </div>
  </div>
</div>
`;
